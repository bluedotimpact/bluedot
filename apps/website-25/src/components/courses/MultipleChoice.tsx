import CTALinkOrButton from '@bluedot/ui/src/CTALinkOrButton';
import React from 'react';

type MultipleChoiceProps = {
  // Required
  title: string;
  question: string;
  options: string[];
  correctOption: string;
};

const MultipleChoice: React.FC<MultipleChoiceProps> = ({
  title,
  question,
  options,
  correctOption,
}) => {
  const [selectedOption, setSelectedOption] = React.useState<string | null>(null);
  const [isCorrect, setIsCorrect] = React.useState<string | null>(null);
  const [isIncorrect, setIsIncorrect] = React.useState<string | null>(null);

  const isSelected = (option: string): boolean => {
    return selectedOption === option;
  };

  const isOptionCorrect = (option: string): boolean => {
    return isCorrect === option;
  };

  const isOptionIncorrect = (option: string): boolean => {
    return isIncorrect === option;
  };

  const answerCheck = () => {
    if (selectedOption === correctOption) {
      setIsCorrect(selectedOption);
      setIsIncorrect(null);
    } else {
      setIsIncorrect(selectedOption);
      setIsCorrect(null);
    }
  };

  return (
    <div className="multiple-choice container-lined bg-white p-8 flex flex-col gap-6">
        <div className="multiple-choice__header flex flex-col gap-4">
            <div className="multiple-choice__header-icon">
                <img src="/icons/lightning_bolt.svg" className="w-15 h-15" />
            </div>
            <div className="multiple-choice__header-content flex flex-col gap-2">
                <p className="multiple-choice__title subtitle-sm">{title}</p>
                <p className="multiple-choice__question">{question}</p>
            </div>
        </div>
      <div className="multiple-choice__options flex flex-col gap-2">
        {options.map((option) => (
          <label className={`
              multiple-choice__option flex items-center gap-2 p-4 hover:cursor-pointer
              ${isSelected(option) ? 'multiple-choice__option--selected container-active' : 'container-lined'}
              ${isOptionCorrect(option) ? 'multiple-choice__option--correct bg-[#63C96533] border-[#63C965]' : ''}
              ${isOptionIncorrect(option) ? 'multiple-choice__option--incorrect bg-[#FF636333] border-[#FF6363]' : ''}
            `}>
            <input
              className="multiple-choice__input"
              type="radio"
              name="multiple-choice"
              value={option}
              onChange={() => setSelectedOption(option)}
            />
            <span className="multiple-choice__label">{option}</span>
          </label>
        ))}
      </div>
      <CTALinkOrButton
        className="multiple-choice__submit" variant="primary"
        onClick={answerCheck}
      >
        Check
      </CTALinkOrButton>
    </div>
  );
};

export default MultipleChoice;
