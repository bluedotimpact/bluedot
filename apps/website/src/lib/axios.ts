import axios from 'axios';
import { configure } from 'axios-hooks';

// Create axios instance with base URL
const axiosInstance = axios.create({
  baseURL: typeof window !== 'undefined' ? window.location.origin : 'http://localhost:8000',
});

// Configure axios-hooks to use our configured instance
configure({ axios: axiosInstance });

export default axiosInstance;
