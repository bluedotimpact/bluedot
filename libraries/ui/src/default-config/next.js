/**
 * @param {import('next').NextConfig & { headers: Header[] }} [config]
 * @returns {import('next').NextConfig}
 */
const withDefaultBlueDotNextConfig = async (config) => ({
  reactStrictMode: true,
  output: 'standalone',
  distDir: 'dist',
  poweredByHeader: false,
  // We already run eslint as a separate step
  eslint: {
    ignoreDuringBuilds: true,
  },
  devIndicators: false,

  ...config,

  transpilePackages: ['@bluedot/ui', ...(config?.transpilePackages ?? [])],
  headers: async () => {
    return [{
      source: '/:path*',
      headers: [
        {
          key: 'X-BlueDot-Version',
          // eslint-disable-next-line turbo/no-undeclared-env-vars
          value: process.env.VERSION_TAG || 'unknown',
        },
      ],
    },
    {
      source: '/api/:path*',
      headers: [
        {
          key: 'Access-Control-Allow-Origin',
          value: '*',
        },
        {
          key: 'Access-Control-Allow-Methods',
          value: 'GET, POST, PUT, PATCH, DELETE, OPTIONS',
        },
        {
          key: 'Access-Control-Allow-Headers',
          value: 'Content-Type, Authorization',
        },
      ],
    },
    ...(config?.headers ?? [])];
  },
  webpack: (webpackConfig, context, ...args) => {
    // Exclude packages from webpack bundling

    // Only used server side. Trying to import it client-side results in an error
    if (!context.isServer) {
      webpackConfig.externals.push('winston');
    }

    // https://github.com/open-telemetry/opentelemetry-js/issues/4173
    // eslint-disable-next-line no-param-reassign
    webpackConfig.ignoreWarnings = webpackConfig.ignoreWarnings ?? [];
    webpackConfig.ignoreWarnings.push({ module: /opentelemetry/, message: /the request of a dependency is an expression/ });

    // Conditionally reuqired, but we don't actually use it
    webpackConfig.externals.push('@opentelemetry/winston-transport');
    webpackConfig.externals.push('@opentelemetry/exporter-jaeger');

    return config?.webpack?.(config, context, ...args) ?? webpackConfig;
  },
});

module.exports = { withDefaultBlueDotNextConfig };
