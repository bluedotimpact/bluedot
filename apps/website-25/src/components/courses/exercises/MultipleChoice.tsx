import { addQueryParam, CTALinkOrButton } from '@bluedot/ui';
import clsx from 'clsx';
import React, { useCallback, useEffect } from 'react';
import { useForm } from 'react-hook-form';
import { ROUTES } from '../../../lib/routes';

type MultipleChoiceProps = {
  // Required
  answer: string;
  description: string;
  onExerciseSubmit: (savedExerciseResponse: string, completed?: boolean) => void;
  options: string;
  title: string;
  // Optional
  className?: string;
  exerciseResponse?: string;
  isLoggedIn?: boolean;
};

type FormData = {
  answer: string;
};

const MultipleChoice: React.FC<MultipleChoiceProps> = ({
  answer,
  className,
  description,
  exerciseResponse,
  isLoggedIn,
  onExerciseSubmit,
  options,
  title,
}) => {
  /**
   * Options are stored as a string with newlines
   * Format them to be an array of strings with no empty strings (i.e., removing trailing return statements)
   */
  const formattedOptions = (options.split('\n').map((o) => o.trim()).filter((o) => o !== ''));
  const formattedAnswer = answer.trim();
  const formattedExerciseResponse = exerciseResponse?.trim();

  const [selectedOption, setSelectedOption] = React.useState<string | null>();
  const [isEditing, setIsEditing] = React.useState<boolean>(false);
  const { register, handleSubmit, setValue } = useForm<FormData>({
    defaultValues: {
      answer: formattedExerciseResponse || '',
    },
  });

  const handleOptionSelect = (option: string) => {
    setSelectedOption(option);
    setIsEditing(true);
  };

  const isSelected = (option: string): boolean => {
    if (!isEditing && formattedExerciseResponse) {
      return formattedExerciseResponse === option;
    }
    return selectedOption === option;
  };

  useEffect(() => {
    if (formattedExerciseResponse) {
      setValue('answer', formattedExerciseResponse);
    }
  }, [formattedExerciseResponse, setValue]);

  const onSubmit = useCallback(async (data: FormData) => {
    try {
      const isAnswerCorrect = data.answer === formattedAnswer;
      console.log('submitted this answer: ', data.answer, ' and is it correct? ', isAnswerCorrect);
      await onExerciseSubmit(data.answer, isAnswerCorrect);
      setIsEditing(false);
    } catch (e) {
      console.log('error', e);
      // Set some state for the error and render it
    }
  }, [onExerciseSubmit]);

  const isCorrect = formattedExerciseResponse && formattedExerciseResponse === formattedAnswer;
  const isIncorrect = formattedExerciseResponse && formattedExerciseResponse !== formattedAnswer;

  return (
    <form onSubmit={handleSubmit(onSubmit)} className={clsx('multiple-choice container-lined bg-white p-8 flex flex-col gap-6', className)}>
      <div className="multiple-choice__header flex flex-col gap-4">
        <div className="multiple-choice__header-icon">
          <img src="/icons/lightning_bolt.svg" className="w-15 h-15" alt="" />
        </div>
        <div className="multiple-choice__header-content flex flex-col gap-2">
          <p className="multiple-choice__title subtitle-sm">{title}</p>
          <p className="multiple-choice__description">{description}</p>
        </div>
      </div>
      <div className="multiple-choice__options flex flex-col gap-2">
        {formattedOptions.map((option) => (
          <label
            key={option}
            className={`
              multiple-choice__option flex items-center gap-2 p-4 hover:cursor-pointer
              ${(isSelected(option)) ? `multiple-choice__option--selected container-active
                ${(!isEditing && isCorrect) && 'multiple-choice__option--correct bg-[#63C96533] border-[#63C965]'}
                ${(!isEditing && isIncorrect) && 'multiple-choice__option--incorrect bg-[#FF636333] border-[#FF6363]'}`
              : 'container-lined'}`}
          >
            <input
              {...register('answer')}
              defaultChecked={formattedExerciseResponse === option}
              className="multiple-choice__input"
              type="radio"
              value={option}
              onChange={() => handleOptionSelect(option)}
              disabled={!isLoggedIn}
            />
            <span className="multiple-choice__label">{option}</span>
          </label>
        ))}
      </div>
      {isLoggedIn ? (
        <CTALinkOrButton
          className="multiple-choice__submit"
          variant="primary"
          type="submit"
        >
          Check
        </CTALinkOrButton>
      ) : (
        <CTALinkOrButton
          className="multiple-choice__login-cta"
          variant="primary"
          url={addQueryParam(ROUTES.login.url, 'redirect_to', window.location.href)}
          withChevron
        >
          Login to check your answer
        </CTALinkOrButton>
      )}
      {(!isEditing && isCorrect) && <p className="multiple-choice__correct-msg">Correct! Quiz completed. ðŸŽ‰</p>}
      {(!isEditing && isIncorrect) && <p className="multiple-choice__incorrect-msg">Try again. ðŸ¤”</p>}
    </form>
  );
};

export default MultipleChoice;
