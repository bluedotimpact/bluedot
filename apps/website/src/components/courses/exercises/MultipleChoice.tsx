import { addQueryParam, CTALinkOrButton, Input } from '@bluedot/ui';
import clsx from 'clsx';
import React, { useCallback, useEffect } from 'react';
import { useForm } from 'react-hook-form';
import { ROUTES } from '../../../lib/routes';
import { P } from '../../Text';
import { formatStringToArray } from '../../../lib/utils';
// eslint-disable-next-line import/no-cycle
import MarkdownExtendedRenderer from '../MarkdownExtendedRenderer';

type MultipleChoiceProps = {
  // Required
  answer: string;
  description: string;
  onExerciseSubmit: (savedExerciseResponse: string, completed?: boolean) => void;
  options: string;
  title: string;
  // Optional
  className?: string;
  exerciseResponse?: string;
  isLoggedIn?: boolean;
};

type FormData = {
  answer: string;
};

const MultipleChoice: React.FC<MultipleChoiceProps> = ({
  answer,
  className,
  description,
  exerciseResponse,
  isLoggedIn,
  onExerciseSubmit,
  options,
  title,
}) => {
  /**
   * Options are stored as a string with newlines
   * Format them to be an array of strings with no empty strings (i.e., removing trailing return statements)
   */
  const formattedOptions = formatStringToArray(options, '\n');
  const formattedAnswer = answer.trim();
  const formattedExerciseResponse = exerciseResponse?.trim();

  const [selectedOption, setSelectedOption] = React.useState<string | null>();
  const [isEditing, setIsEditing] = React.useState<boolean>(false);
  const {
    register, handleSubmit, setValue, formState: { isSubmitting },
  } = useForm<FormData>({
    defaultValues: {
      answer: formattedExerciseResponse || '',
    },
  });

  const handleOptionSelect = (option: string) => {
    setSelectedOption(option);
    setIsEditing(true);
  };

  const isSelected = (option: string): boolean => {
    if (!isEditing && formattedExerciseResponse) {
      return formattedExerciseResponse === option;
    }
    return selectedOption === option;
  };

  useEffect(() => {
    if (formattedExerciseResponse) {
      setValue('answer', formattedExerciseResponse);
    }
  }, [formattedExerciseResponse, setValue]);

  const onSubmit = useCallback(async (data: FormData) => {
    const isAnswerCorrect = data.answer === formattedAnswer;
    await onExerciseSubmit(data.answer, isAnswerCorrect);
    setIsEditing(false);
  }, [onExerciseSubmit]);

  const isCorrect = formattedExerciseResponse && formattedExerciseResponse === formattedAnswer;
  const isIncorrect = formattedExerciseResponse && formattedExerciseResponse !== formattedAnswer;

  return (
    <form onSubmit={handleSubmit(onSubmit)} className={clsx('multiple-choice container-lined bg-white p-8 flex flex-col gap-6', className)}>
      <div className="multiple-choice__header flex flex-col gap-4">
        <div className="multiple-choice__header-icon not-prose">
          <img src="/icons/lightning_bolt.svg" className="w-15 h-15" alt="" />
        </div>
        <div className="multiple-choice__header-content flex flex-col gap-2">
          <p className="multiple-choice__title bluedot-h4 not-prose">{title}</p>
          <MarkdownExtendedRenderer className="multiple-choice__description">{description}</MarkdownExtendedRenderer>
        </div>
      </div>
      <div className="multiple-choice__options flex flex-col gap-2">
        {formattedOptions.map((option) => (
          <Input
            {...register('answer')}
            labelClassName={`
              multiple-choice__option flex items-center gap-2 p-4 hover:cursor-pointer
              ${(isSelected(option)) ? `multiple-choice__option--selected container-active
                ${(!isEditing && isCorrect) && 'multiple-choice__option--correct bg-[#63C96533] border-[#63C965]'}
                ${(!isEditing && isIncorrect) && 'multiple-choice__option--incorrect bg-[#FF636333] border-[#FF6363]'}`
              : 'container-lined'}`}
            defaultChecked={formattedExerciseResponse === option}
            type="radio"
            value={option}
            onChange={() => handleOptionSelect(option)}
            disabled={!isLoggedIn}
          />
        ))}
      </div>
      {isLoggedIn ? (
        <CTALinkOrButton
          className="multiple-choice__submit"
          variant="primary"
          onClick={handleSubmit(onSubmit)}
          disabled={isSubmitting}
        >
          {isSubmitting ? 'Checking...' : 'Check'}
        </CTALinkOrButton>
      ) : (
        <CTALinkOrButton
          className="multiple-choice__login-cta"
          variant="primary"
          url={addQueryParam(addQueryParam(ROUTES.login.url, 'redirect_to', window.location.pathname), 'register', 'true')}
          withChevron
        >
          Create a free account to check your answer
        </CTALinkOrButton>
      )}
      {(!isEditing && isCorrect) && <P className="multiple-choice__correct-msg">Correct! Quiz completed. ðŸŽ‰</P>}
      {(!isEditing && isIncorrect) && <P className="multiple-choice__incorrect-msg">Try again. ðŸ¤”</P>}
    </form>
  );
};

export default MultipleChoice;
