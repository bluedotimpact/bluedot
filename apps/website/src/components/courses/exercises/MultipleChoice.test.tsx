import '@testing-library/jest-dom';
import { render, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import axios from 'axios';
import { useRouter } from 'next/router';
import {
  beforeEach, describe, expect, Mock, test, vi,
} from 'vitest';
import MultipleChoice from './MultipleChoice';

// Mock next/router
vi.mock('next/router', () => ({
  useRouter: vi.fn(),
}));

// Mock axios
vi.mock('axios');
(axios.put as Mock).mockResolvedValue({ data: {} });

const mockRouter = {
  asPath: '/test-path',
};

// Setup router mock before each test
beforeEach(() => {
  (useRouter as Mock).mockReturnValue(mockRouter);
});

const mockOptions = "The community's preference for low-tech fishing traditions\nRising consumer demand for fish with more Omega-3s\nEnvironmental regulations and declining cod stocks\nA cultural shift toward vegetarianism in the region\n";

const mockArgs = {
  title: 'Understanding LLMs',
  description:
    "Why is a language model's ability to predict 'the next word' capable of producing complex behaviors like solving maths problems?",
  options: mockOptions,
  answer: "The community's preference for low-tech fishing traditions\n",
  onExerciseSubmit: () => {},
};

describe('MultipleChoice', () => {
  test('renders default as expected', async () => {
    const { container, getByText, getAllByRole } = render(<MultipleChoice {...mockArgs} />);

    // Wait for MarkdownExtendedRenderer to complete async rendering
    await waitFor(() => {
      expect(getByText(mockArgs.description)).toBeInTheDocument();
    });

    expect(container).toMatchSnapshot();

    // Expect no options to be selected
    const radioInputs = getAllByRole('radio');
    radioInputs.forEach((input) => {
      expect(input).not.toBeChecked();
      // All inputs are disabled until user is logged in
      expect(input).toBeDisabled();
    });
  });

  test('renders logged in as expected', async () => {
    const { container, getByText, getAllByRole } = render(<MultipleChoice {...mockArgs} isLoggedIn />);

    // Wait for MarkdownExtendedRenderer to complete async rendering
    await waitFor(() => {
      expect(getByText(mockArgs.description)).toBeInTheDocument();
    });

    expect(container).toMatchSnapshot();

    // Expect no options to be selected
    const radioInputs = getAllByRole('radio');
    radioInputs.forEach((input) => {
      expect(input).not.toBeChecked();
      expect(input).toBeEnabled();
    });
  });

  test('disables submit button when no option selected', () => {
    const { getByRole } = render(<MultipleChoice {...mockArgs} isLoggedIn />);

    const submitButton = getByRole('button', { name: /select an option/i });
    expect(submitButton).toBeDisabled();
  });

  test('enables submit button when option selected', async () => {
    const user = userEvent.setup();
    const { getAllByRole, getByRole } = render(<MultipleChoice {...mockArgs} isLoggedIn />);

    const radioInputs = getAllByRole('radio');
    if (!radioInputs[0]) throw new Error('No radio inputs found');
    await user.click(radioInputs[0]);

    const submitButton = getByRole('button', { name: /check answer/i });
    expect(submitButton).toBeEnabled();
  });

  test('updates styles for selected option', async () => {
    const user = userEvent.setup();
    const { getAllByRole } = render(<MultipleChoice {...mockArgs} isLoggedIn />);
    // Select the first option
    const radioInputs = getAllByRole('radio');
    const firstOption = radioInputs[0];
    if (!firstOption) throw new Error('No radio input found');

    await user.click(firstOption);

    // Expect first radio to be checked, and others not
    expect(firstOption).toBeChecked();
    radioInputs.slice(1).forEach((input) => {
      expect(input).not.toBeChecked();
    });
  });

  test('updates styles for correct option', async () => {
    const { getByDisplayValue, getByText } = render(
      <MultipleChoice {...mockArgs} exerciseResponse={mockArgs.answer} isLoggedIn />,
    );

    // Wait for MarkdownExtendedRenderer to complete async rendering
    await waitFor(() => {
      expect(getByText(mockArgs.description)).toBeInTheDocument();
    });

    const correctRadio = getByDisplayValue(mockArgs.answer.trim());
    expect(correctRadio).toBeChecked();
    expect(correctRadio).toMatchSnapshot();
    expect(getByText('Correct! Quiz completed. ðŸŽ‰')).toBeInTheDocument();
  });

  test('updates styles for incorrect option', async () => {
    const incorrectAnswer = 'Rising consumer demand for fish with more Omega-3s\n';
    const { getByDisplayValue, getByRole, getByText } = render(
      <MultipleChoice {...mockArgs} exerciseResponse={incorrectAnswer} isLoggedIn />,
    );

    // Wait for MarkdownExtendedRenderer to complete async rendering
    await waitFor(() => {
      expect(getByText(mockArgs.description)).toBeInTheDocument();
    });

    const incorrectRadio = getByDisplayValue(incorrectAnswer.trim());
    expect(incorrectRadio).toBeChecked();
    expect(incorrectRadio).toMatchSnapshot();

    const tryAgainButton = getByRole('button', { name: /try again/i });
    expect(tryAgainButton).toBeInTheDocument();
    expect(tryAgainButton).toMatchSnapshot();
  });
});
