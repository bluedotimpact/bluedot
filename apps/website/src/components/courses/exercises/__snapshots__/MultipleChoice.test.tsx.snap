// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MultipleChoice > renders default as expected 1`] = `
<div>
  <form
    class="multiple-choice container-lined bg-white p-8 flex flex-col gap-6"
  >
    <div
      class="multiple-choice__header flex flex-col gap-4"
    >
      <div
        class="multiple-choice__header-icon"
      >
        <img
          alt=""
          class="w-15 h-15"
          src="/icons/lightning_bolt.svg"
        />
      </div>
      <div
        class="multiple-choice__header-content flex flex-col gap-2"
      >
        <p
          class="multiple-choice__title bluedot-h4"
        >
          Understanding LLMs
        </p>
        <p
          class="bluedot-p multiple-choice__description"
        >
          Why is a language model's ability to predict 'the next word' capable of producing complex behaviors like solving maths problems?
        </p>
      </div>
    </div>
    <div
      class="multiple-choice__options flex flex-col gap-2"
    >
      <label
        class="
              multiple-choice__option flex items-center gap-2 p-4 hover:cursor-pointer
              container-lined"
      >
        <input
          class="multiple-choice__input"
          disabled=""
          name="answer"
          type="radio"
          value="The community's preference for low-tech fishing traditions"
        />
        <span
          class="multiple-choice__label"
        >
          The community's preference for low-tech fishing traditions
        </span>
      </label>
      <label
        class="
              multiple-choice__option flex items-center gap-2 p-4 hover:cursor-pointer
              container-lined"
      >
        <input
          class="multiple-choice__input"
          disabled=""
          name="answer"
          type="radio"
          value="Rising consumer demand for fish with more Omega-3s"
        />
        <span
          class="multiple-choice__label"
        >
          Rising consumer demand for fish with more Omega-3s
        </span>
      </label>
      <label
        class="
              multiple-choice__option flex items-center gap-2 p-4 hover:cursor-pointer
              container-lined"
      >
        <input
          class="multiple-choice__input"
          disabled=""
          name="answer"
          type="radio"
          value="Environmental regulations and declining cod stocks"
        />
        <span
          class="multiple-choice__label"
        >
          Environmental regulations and declining cod stocks
        </span>
      </label>
      <label
        class="
              multiple-choice__option flex items-center gap-2 p-4 hover:cursor-pointer
              container-lined"
      >
        <input
          class="multiple-choice__input"
          disabled=""
          name="answer"
          type="radio"
          value="A cultural shift toward vegetarianism in the region"
        />
        <span
          class="multiple-choice__label"
        >
          A cultural shift toward vegetarianism in the region
        </span>
      </label>
    </div>
    <a
      class="cta-button flex items-center justify-center rounded-sm transition-all duration-200 text-sm px-4 py-3 w-fit font-[650] whitespace-nowrap cursor-pointer cta-button--primary bg-bluedot-normal link-on-dark multiple-choice__login-cta"
      href="http://localhost:3000/login?redirect_to=http%3A%2F%2Flocalhost%3A3000%2F"
      tabindex="0"
    >
      <span
        class="cta-button__text"
      >
        Login to check your answer
      </span>
      <span
        class="cta-button__chevron ml-3"
      >
        <svg
          class="cta-button__chevron-icon size-2"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 320 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M310.6 233.4c12.5 12.5 12.5 32.8 0 45.3l-192 192c-12.5 12.5-32.8 12.5-45.3 0s-12.5-32.8 0-45.3L242.7 256 73.4 86.6c-12.5-12.5-12.5-32.8 0-45.3s32.8-12.5 45.3 0l192 192z"
          />
        </svg>
      </span>
    </a>
  </form>
</div>
`;

exports[`MultipleChoice > renders logged in as expected 1`] = `
<div>
  <form
    class="multiple-choice container-lined bg-white p-8 flex flex-col gap-6"
  >
    <div
      class="multiple-choice__header flex flex-col gap-4"
    >
      <div
        class="multiple-choice__header-icon"
      >
        <img
          alt=""
          class="w-15 h-15"
          src="/icons/lightning_bolt.svg"
        />
      </div>
      <div
        class="multiple-choice__header-content flex flex-col gap-2"
      >
        <p
          class="multiple-choice__title bluedot-h4"
        >
          Understanding LLMs
        </p>
        <p
          class="bluedot-p multiple-choice__description"
        >
          Why is a language model's ability to predict 'the next word' capable of producing complex behaviors like solving maths problems?
        </p>
      </div>
    </div>
    <div
      class="multiple-choice__options flex flex-col gap-2"
    >
      <label
        class="
              multiple-choice__option flex items-center gap-2 p-4 hover:cursor-pointer
              container-lined"
      >
        <input
          class="multiple-choice__input"
          name="answer"
          type="radio"
          value="The community's preference for low-tech fishing traditions"
        />
        <span
          class="multiple-choice__label"
        >
          The community's preference for low-tech fishing traditions
        </span>
      </label>
      <label
        class="
              multiple-choice__option flex items-center gap-2 p-4 hover:cursor-pointer
              container-lined"
      >
        <input
          class="multiple-choice__input"
          name="answer"
          type="radio"
          value="Rising consumer demand for fish with more Omega-3s"
        />
        <span
          class="multiple-choice__label"
        >
          Rising consumer demand for fish with more Omega-3s
        </span>
      </label>
      <label
        class="
              multiple-choice__option flex items-center gap-2 p-4 hover:cursor-pointer
              container-lined"
      >
        <input
          class="multiple-choice__input"
          name="answer"
          type="radio"
          value="Environmental regulations and declining cod stocks"
        />
        <span
          class="multiple-choice__label"
        >
          Environmental regulations and declining cod stocks
        </span>
      </label>
      <label
        class="
              multiple-choice__option flex items-center gap-2 p-4 hover:cursor-pointer
              container-lined"
      >
        <input
          class="multiple-choice__input"
          name="answer"
          type="radio"
          value="A cultural shift toward vegetarianism in the region"
        />
        <span
          class="multiple-choice__label"
        >
          A cultural shift toward vegetarianism in the region
        </span>
      </label>
    </div>
    <button
      class="cta-button flex items-center justify-center rounded-sm transition-all duration-200 text-sm px-4 py-3 w-fit font-[650] whitespace-nowrap cursor-pointer cta-button--primary bg-bluedot-normal link-on-dark multiple-choice__submit"
      type="button"
    >
      <span
        class="cta-button__text"
      >
        Check
      </span>
    </button>
  </form>
</div>
`;

exports[`MultipleChoice > updates styles for correct option 1`] = `
<label
  class="
              multiple-choice__option flex items-center gap-2 p-4 hover:cursor-pointer
              multiple-choice__option--selected container-active
                multiple-choice__option--correct bg-[#63C96533] border-[#63C965]
                false"
>
  <input
    class="multiple-choice__input"
    name="answer"
    type="radio"
    value="The community's preference for low-tech fishing traditions"
  />
  <span
    class="multiple-choice__label"
  >
    The community's preference for low-tech fishing traditions
  </span>
</label>
`;

exports[`MultipleChoice > updates styles for correct option 2`] = `
<p
  class="bluedot-p multiple-choice__correct-msg"
>
  Correct! Quiz completed. ðŸŽ‰
</p>
`;

exports[`MultipleChoice > updates styles for correct option 3`] = `
<label
  class="
              multiple-choice__option flex items-center gap-2 p-4 hover:cursor-pointer
              multiple-choice__option--selected container-active
                false
                multiple-choice__option--incorrect bg-[#FF636333] border-[#FF6363]"
>
  <input
    class="multiple-choice__input"
    name="answer"
    type="radio"
    value="Rising consumer demand for fish with more Omega-3s"
  />
  <span
    class="multiple-choice__label"
  >
    Rising consumer demand for fish with more Omega-3s
  </span>
</label>
`;

exports[`MultipleChoice > updates styles for correct option 4`] = `
<p
  class="bluedot-p multiple-choice__incorrect-msg"
>
  Try again. ðŸ¤”
</p>
`;
